src/ORM/DataExtension.php

```diff
@@ -60,22 +60,52 @@ public function augmentWrite(&$manipulation)
     {
     }
 
+    /**
+     * Optional extension hook called by SilverStripe directly before a database write operation.
+     * Any logic expressed here, will be executed by SilverStripe.
+     *
+     * @return void
+     */
     public function onBeforeWrite()
     {
     }
 
+    /**
+     * Optional extension hook called by SilverStripe directly after a database write operation.
+     * Any logic expressed here, will be executed by SilverStripe.
+     *
+     * @return void
+     */
     public function onAfterWrite()
     {
     }
 
+    /**
+     * Optional extension hook called by SilverStripe directly before deleting a database write operation.
```

Comment:
>directly before deleting a database write operation

I think this might be a result of a copy/paste, would you mind updating it to say something like "directly before deleting a DataObject" or something like that?